concent_api/middleman/utils.py

```diff
@@ -0,0 +1,78 @@
+from typing import Dict
+
+import asyncio
+
+from logging import getLogger
+from logging import Logger
+
+from middleman.constants import ResponseQueueItem
+
+STANDARD_ERROR_MESSAGE = "This connection has been already added"
+WRONG_TYPE_ERROR_MESSAGE = "QueuePool can only contain mapping between connection ID (int) and response queue (asyncio.Queue)"
+
+
+class QueuePool(dict):
+    def __init__(
+        self,
+        initial_data: Dict[int, asyncio.Queue] = None,
+        loop: asyncio.BaseEventLoop = None,
+        logger: Logger = None
+    ) -> None:
+        super(QueuePool, self).__init__({k: v for k, v in initial_data.items()} if initial_data is not None else {})
+        self.logger = logger if logger is not None else getLogger()
+        self.loop = loop if loop is not None else asyncio.get_event_loop()
+        for key, value in self.items():
+            self._validate_mapping(key, value)
+
+    def __setitem__(self, key, value):
+        self._validate_mapping(key, value)
+        self._ensure_key_uniqueness(key)
+        super(QueuePool, self).__setitem__(key, value)
+
+    def _ensure_key_uniqueness(self, key):
+        if key in self:
+            raise KeyError(STANDARD_ERROR_MESSAGE)
+
+    @staticmethod
```

Comment:
Why static method ?

Reply:
no self use -> it is a static method. Or, if you prefer, I can make a function out of it.

Reply:
done